acvm/src/pwg/directives.rs

```diff
@@ -90,52 +90,44 @@ pub fn solve_directives(
-                    return Err(OpcodeResolutionError::UnsatisfiedConstrain);
-                }
+            // If the number of witnesses provided is less than the number of
+            // decomposed integers then we return an error.
+            //
+            // If the number of witnesses provided is more, then we pad
+            // the decomposed integer with zeroes to assign the extra
+            // witnesses to zero.
+            if b.len() < decomposed_integer.len() {
+                return Err(OpcodeResolutionError::UnsatisfiedConstrain);
+            }
+            let padding_len = b.len() - decomposed_integer.len();
+            let zeroes = vec![0u8; padding_len];
 
-                for (i, witness) in b.iter().enumerate() {
-                    // Fetch the `i'th` digit from the decomposed integer list
-                    // and convert it to a field element.
-                    // If it is not available, which can happen when the decomposed integer
-                    // list is shorter than the witness list, we return 0.
-                    let value = match decomposed_integer.get(i) {
-                        Some(digit) => FieldElement::from_be_bytes_reduce(&[*digit]),
-                        None => FieldElement::zero(),
-                    };
-
-                    insert_value(witness, value, initial_witness)?
-                }
+            let padded_decomposed_integer: Vec<_> = if *is_little_endian {
+                // Pad the beginning with zeroes for little endian
+                zeroes.into_iter().chain(decomposed_integer).collect()
             } else {
-                // Decompose the integer into its radix digits in big endian form.
-                let decomposed_integer = big_integer.to_radix_be(*radix);
-
-                // if it is big endian and the decompoased integer list is shorter
-                // than the witness list, pad the extra part with 0 first then
-                // add the decompsed interger list to the witness list.
-                let padding_len = b.len() - decomposed_integer.len();
-                let mut value = FieldElement::zero();
-                for (i, witness) in b.iter().enumerate() {
-                    if i >= padding_len {
-                        value = match decomposed_integer.get(i - padding_len) {
-                            Some(digit) => FieldElement::from_be_bytes_reduce(&[*digit]),
-                            None => {
-                                return Err(OpcodeResolutionError::OpcodeNotSolvable(
-                                    OpcodeNotSolvable::UnreachableCode,
-                                ))
-                            }
-                        };
-                    }
-                    insert_value(witness, value, initial_witness)?
-                }
+                // Pad the ending with zeroes for big endian
+                decomposed_integer.into_iter().chain(zeroes).collect()
```

Comment:
Should this not be switched? If I have `0x3B4C` in big endian and I am padding `0000`, I want `0x00003B4C` not `0x3B4C0000`. While with LE I would want `0x4C3B0000`

Reply:
Yes so for little endian we push to the right and for big endian we prepend to the left.

I think this is what we were doing before (thinking)